base/reducedim.jl

```diff
@@ -238,7 +238,7 @@ copyfirst!(R::AbstractArray, A::AbstractArray) = mapfirst!(identity, R, A)
 function mapfirst!(f::F, R::AbstractArray, A::AbstractArray{<:Any,N}) where {N, F}
     lsiz = check_reducedims(R, A)
     t = _firstreducedslice(axes(R), axes(A))
-    map!(f, R, view(A, t...))
+    map!(f, view(R, firstindex(R):lastindex(R)), view(A, t...))
```

Comment:
Why did this need to change?

```suggestion
    map!(f, R, view(A, t...))
```

Reply:
otherwise it will cause `maximum!` and `minimum!` on `OffsetArrays` to fail via some interaction with `mapfirst!` since `view(R::OffsetArray, :)` has offset indices but `view(R::OffsetArray, firstindex(R):lastindex(R))` has 1-based indices.